bilm/data.py

```diff
@@ -158,7 +158,7 @@ def _convert_word_to_char_ids(self, word):
         code[0] = self.bow_char
         for k, chr_id in enumerate(word_encoded, start=1):
             code[k] = chr_id
-        code[k + 1] = self.eow_char
+        code[len(word_encoded)] = self.eow_char
```

Comment:
This should only be necessary if you have empty strings `""` in your vocabulary.  Besides, it is a breaking change and incompatible with existing trained models.